state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -879,46 +862,57 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	/// The broadcast will then be retried.
 	fn handle_broadcast_failure(
 		broadcast_id: BroadcastId,
-		maybe_reporter: Option<T::ValidatorId>,
+		mut maybe_reporter: Option<T::ValidatorId>,
 	) -> DispatchResult {
-		let signing_attempt = AwaitingBroadcast::<T, I>::get(broadcast_id)
+		let broadcast = AwaitingBroadcast::<T, I>::get(broadcast_id)
 			.ok_or(Error::<T, I>::InvalidBroadcastId)?;
 
-		let failed_broadcaster = maybe_reporter.unwrap_or(signing_attempt.nominee);
+		if maybe_reporter.is_none() {
+			maybe_reporter = broadcast.nominee;
+		}
 
-		// NOTE: We don't use `append` here because we want to make sure not to insert duplicates
-		// and `append` doesn't guarantee that.
```

Comment:
I think this comment is useful to avoid someone trying to optimise this in the future, by using StorageAppend::append.

Reply:
Now that we use "set", it is absolutely fine to use append, so this comment is now redandunt.

Reply:
the storage append takes in an individual item, whereas BTreeSet append takes a BTreeSet, so both can be used when appropriate.

Reply:
The issue with StorageAppend is that it causes `decode_len` to become inconsistent.

For example if you do this: 

```rust
// Use StorageAppend::append
FailedBroadcasters::<T>::append(Default::default());
FailedBroadcasters::<T>::append(Default::default());
let count = FailedBroadcasters::<T>::decode_len();
```

`count` will be 2.

But with this:

```rust
// Use BTreeSet::insert
FailedBroadcasters::<T>::mutate(|vals| vals.insert(Default::default()));
FailedBroadcasters::<T>::mutate(|vals| vals.insert(Default::default()));
let count = FailedBroadcasters::<T>::decode_len();
```

`count` will be 1.


Maybe the comment wasn't very clear, but the intention was that if someone notices that we use the second method (which is much less efficient), they might be tempted to change it to the first, but this would introduce a potential bug.